### PR TITLE
compound index added for survey and observation

### PR DESCRIPTION
**Pull request by priyanka-TL**

### CODE REVIEW

File: models/surveys.js [hunk 1]
```diff
@@ -43,5 +43,11 @@ module.exports = {
         default : false,
         type : Boolean
       }
-    }
+    },
+    runIndex: [
```
Comment (aks30) [inline models/surveys.js:47]: @priyanka-TL Rename this to compoundIndex

File: models/observationSubmissions.js [hunk 2]
```diff
@@ -90,5 +90,11 @@ module.exports = {
     userRoleInformation : Object,
     criteriaLevelReport : Boolean,
     userProfile : Object
-  }
+  },
+  compoundIndex: [
+    {
+      "name" :{ createdBy: 1, observationId: 1, submissionNumber:1 },
```
Comment (aks30) [inline models/observationSubmissions.js:96]: @priyanka-TL i think this should have observationId: 1, submissionNumber:1  and entityId :1

File: models/observations.js [hunk 3]
```diff
@@ -65,5 +65,11 @@ module.exports = {
       type : Boolean,
       index: true
     }
-  }
+  },
+  compoundIndex: [
+    {
+        "name" :{ createdBy: 1, solutionId: 1 },
+        "indexType" : { unique: true, partialFilterExpression: { solutionId: { $exists: true }}}
```
Comment (aks30) [inline models/observations.js:72]: @priyanka-TL there will never be a case where solutionId is not there right?

Comment (priyanka-TL) [reply models/observations.js:72]: no

File: models/observationSubmissions.js (hunk 2 above)
Comment (priyanka-TL) [reply models/observationSubmissions.js:96]: when we observe again entityId can be same. For same entity also we can observe again

File: models/surveySubmissions.js [hunk 4]
```diff
@@ -50,6 +50,12 @@ module.exports = {
       appInformation : Object,
       userRoleInformation : Object,
       userProfile : Object
-    }
+    },
+    compoundIndex: [
```
Comment (aks30) [inline models/surveySubmissions.js:54]: @priyanka-TL No need just make surveyId unique

File: models/surveys.js [hunk 5]
```diff
@@ -43,5 +43,11 @@ module.exports = {
         default : false,
         type : Boolean
       }
-    }
+    },
+    compoundIndex: [
+      {
+          "name" :{ createdBy: 1, solutionId: 1 },
+          "indexType" : { unique: true, partialFilterExpression: { solutionId: { $exists: true }}}
```
Comment (aks30) [inline models/surveys.js:50]: @priyanka-TL this is not required

**Pull request merged**